apps/test/unit/gamelab/GameLabSpriteTest.js

```diff
@@ -430,6 +423,128 @@ describe('GameLabSprite', function () {
+    // If the animation is currently playing, this has no effect.
+    // If the animation is stopped at the last frame, this will restart the
+    // animation from the beginning.
+    // If the animation is stopped at any other frame, this will resume playing
+    // the animation at that frame.
+
+    const LOOPING_ANIMATION = 'looping-animation',
+      NON_LOOPING_ANIMATION = 'non-looping-animation';
+    let sprite;
+
+    beforeEach(function () {
+      sprite = createSprite(0, 0);
+
+      // Manually preload animations onto p5.projectAnimations
+      gameLabP5.p5.projectAnimations = {
+        [LOOPING_ANIMATION]: createTestAnimation(3, true),
+        [NON_LOOPING_ANIMATION]: createTestAnimation(3, false)
+      };
+    });
+    it('has no effect on a playing, looping animation', function () {
+      sprite.setAnimation(LOOPING_ANIMATION);
+      expect(sprite.animation.playing).to.be.true;
+      expect(sprite.animation.getFrame()).to.equal(0);
+
+      sprite.play();
+      expect(sprite.animation.playing).to.be.true;
+      expect(sprite.animation.getFrame()).to.equal(0);
+
+      sprite.update();  // The test animation frameDelay=1, so this advances a frame.
+      expect(sprite.animation.playing).to.be.true;
+      expect(sprite.animation.getFrame()).to.equal(1);
+
+      sprite.play();
+      expect(sprite.animation.playing).to.be.true;
+      expect(sprite.animation.getFrame()).to.equal(1);
+
+      sprite.update();
+      expect(sprite.animation.playing).to.be.true;
+      expect(sprite.animation.getFrame()).to.equal(2);
+
+      sprite.play();
+      expect(sprite.animation.playing).to.be.true;
+      expect(sprite.animation.getFrame()).to.equal(2);
+
+      sprite.update();
+      expect(sprite.animation.playing).to.be.true;
+      expect(sprite.animation.getFrame()).to.equal(0);
+
+      sprite.play();
+      expect(sprite.animation.playing).to.be.true;
+      expect(sprite.animation.getFrame()).to.equal(0);
+    });
+
+    it('has no effect on a playing, non-looping animation until it reaches the final frame', function () {
+      sprite.setAnimation(NON_LOOPING_ANIMATION);
+      expect(sprite.animation.playing).to.be.true;
+      expect(sprite.animation.getFrame()).to.equal(0);
+
+      sprite.play();
+      expect(sprite.animation.playing).to.be.true;
+      expect(sprite.animation.getFrame()).to.equal(0);
+
+      sprite.update();  // The test animation frameDelay=1, so this advances a frame.
+      expect(sprite.animation.playing).to.be.true;
+      expect(sprite.animation.getFrame()).to.equal(1);
+
+      sprite.play();
+      expect(sprite.animation.playing).to.be.true;
+      expect(sprite.animation.getFrame()).to.equal(1);
+
+      sprite.update();
+      expect(sprite.animation.playing).to.be.true;
+      expect(sprite.animation.getFrame()).to.equal(2);
+
+      sprite.play();
+      expect(sprite.animation.playing).to.be.true;
+      expect(sprite.animation.getFrame()).to.equal(0);
```

Comment:
I'm skeptical about that change, just given that it might touch other things. I think what you've done is the right way to go.